### ISSUE TITLE
BUG - Missing elements when redifining array for dom-repeat with mapbox-layer

### ISSUE THREAD

**Issue by ralexrdz** [PolymerVis/mapbox-gl#32]
I have a custom Polymer component where I recieve an array of geojsons which I called `layers`. for Every layer I add inside a `dom-repeat` a `mapbox-layer` and a `geojson-source`
``` html
    <template is="dom-repeat" items="[[layers]]">
        <mapbox-layer
          map="[[map]]"
          layer-id="[[item.id]]"
          rendering-type="[[item.type]]"
          source="[[item.source]]"
          color="[[item.color]]"
          opacity="0.7"
        >
        </mapbox-layer>
        <geojson-source
          map="[[map]]"
          source-id="[[item.source]]"
          source-data="[[item.data]]"
        >
        </geojson-source>
      </template>
```
Everytime that array of geojsons change I would like to remove all layers and redefine my property `layers`.

Map should not show old layers and show the new layers. 

Instead the map is missing the same number of element from the new array as the number of elements in the first array (recentrly removed).

Here is a basic repo I did to reproduce the BUG [mapbox-layers-bug](https://github.com/ralexrdz/mapbox-layers-bug) 

`layers-behabior.js` file should need this changes to work as explained in this other issue #31 

``` js
   // line 430~
   _setFilter: function(map, layerId, filter) {
      if (!map || !layerId || !this._added) return;
      let layer = this.map.getLayer(layerId)
      if (layer) return map.setFilter(layerId, filter);
    },
...
   // line 500~
   _setPaintProperty: function(layerId, field, value, prefix) {
      if (!this._added) return;
      var _field = typeof prefix == 'string' ? prefix + '-' + field : field;
      let layer = this.map.getLayer(layerId)
      if (layer) this.map.setPaintProperty(layerId, _field, value);
    },

    _setLayoutProperty: function(layerId, field, value, prefix) {
      if (!this._added) return;
      var _field = typeof prefix == 'string' ? prefix + '-' + field : field;
      let layer = this.map.getLayer(layerId)
      if (layer) this.map.setLayoutProperty(layerId, _field, value);
    },
```

**Comment by ralexrdz** [comment]
I found out on `mapbox-layer.js` that `this._added` is `true` for those first elements missing so it doesn't trigger the method `this.map.addLayer(this._layer);`

**Comment by eterna2** [comment]
@ralexrdz Thanks for spotting this! I will go look at it. Most likely i need to unset the flag when the layer is removed.

**Comment by eterna2** [comment]
@ralexrdz 

Hi,

I have fixed the bugs for #31 and #32.

Now `mapbox-gl` will properly add, remove, or update the layers when `mapbox-layer` is created dynamically with `dom-repeat`.

PS: You do not need to manually remove the layers anymore. 

For your use case, I would recommend u to insert the geojson directly via the `source-data` property, instead of needing to add both a layer and a geojson source. 

See: https://www.webcomponents.org/element/PolymerVis/mapbox-gl/demo/demo/mapbox-source-data.html

Alternatively, you might want to create your `geojson-source` outside of `dom-repeat` (as in pre-load all your sources). And update ur layers by updating the `source` properties of each layers.

**Comment by eterna2** [comment]
@ralexrdz 

Note that `source-data` takes in a JSON of the form
```js
{
  "type": "geojson", "data": <ACTUAL_GEOJSON_OBJECT>
}
```
